### PR TITLE
Bind remember article on keyboard shortcut.

### PR DESCRIPTION
**Pull request by drets**
Fixes: https://github.com/heutagogy/heutagogy-chrome-extension/issues/9

@Faradey27 as I told you I am not so familiar with react/redux (yet), so I'll be happy to hear from you how to solve the issue in a proper way (so I'll be able to change it).

### CODE REVIEW

**Comment by rasendubi** [comment]
Try to avoid `fixes #xxx` in the future. A good practice is to have the issue author to confirm the problem is solved and close the issue. (We can always close the issue later if this gets too long.)

File: app/store/utils.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import Immutable from 'immutable';
+
+export const callWithStore = (callback) => {
```
Comment (Faradey27) [inline app/store/utils.js:3]: this is good idea, but you combine here part of redux setup and reading from localStorage

better to left here only logic related to localStorage

like create class with methids

getInitialState
setInitialSTate

which will just return data from localStorage


and better to have all redux setup in one place

File: chrome/extension/background/keyBindings.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import Immutable from 'immutable';
+import { callWithStore } from '../../../app/store/utils';
+import { rememberArticle } from '../../../app/actions/article';
+
+callWithStore((store) => {
```
Comment (Faradey27) [inline chrome/extension/background/keyBindings.js:5]: why name is callWithStore?

File: chrome/extension/background/keyBindings.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  chrome.commands.onCommand.addListener((command) => {
+    if (command === 'remember-article') {
+      chrome.tabs.query({
```
Comment (Faradey27) [inline chrome/extension/background/keyBindings.js:8]: better to move this logic to utils and after that reuse it for two cases - when user click on save button(switch) and whe he enter special command(key bindings)

**Comment by Faradey27** [comment]
@rasendubi @drets i left several small comments

File: chrome/extension/background/keyBindings.js (hunk 2 above)
Comment (drets) [reply chrome/extension/background/keyBindings.js:5]: While giving a name to the method I was concentrated on the `callback(store)` part, but you are right method also sets up redux, reads from localStorage...

File: app/store/utils.js (hunk 1 above)
Comment (drets) [reply app/store/utils.js:3]: Makes sense to separate localStorage logic :+1: 

However, I still don't understand how can I get all redux setup in one place.
I need to bind keys logic in “background.js” (before “app.js” == before opening extension window), so `store` should be also created there.
Then, inside “app.js” I need to pass `store` to `<Root store={store} />`.
I didn't find a way how to use previously created store there (I even tried to save it to background window, but then it throws some strange errors) except re-creating it again, calling `createStore` function second time.

Any ideas?

**Pull request merged**